module/VuFind/src/VuFind/Db/Entity/AccessTokenInterface.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Entity model interface for access tokens.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Entity;
+
+/**
+ * Entity model interface for access tokens.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+interface AccessTokenInterface
+{
+    /**
+     * Set user ID.
+     *
+     * @param ?string $userId User ID
+     *
+     * @return void
+     */
+    public function setUserId(?string $userId): void;
+
+    /**
+     * Set data.
+     *
+     * @param string $data Data
+     *
+     * @return void
+     */
+    public function setData(string $data): void;
```

Comment:
It might be useful to have the setters return AccessTokenInterface so we can support a fluent interface.

Reply:
Oops, missed this, will do now.

Reply:
Done!